src/nbl/video/IOpenGLMemoryAllocation.h

```diff
@@ -0,0 +1,58 @@
+#ifndef __NBL_I_OPENGL_MEMORY_ALLOCATION_H_INCLUDED__
+
+#include "nbl/video/IDriverMemoryAllocation.h"
+
+#include <volk.h>
+
+namespace nbl::video
+{
+
+class ILogicalDevice;
+
+class IOpenGLMemoryAllocation : public IDriverMemoryAllocation
+{
+    using base_t = IDriverMemoryAllocation;
+public:
+    IOpenGLMemoryAllocation(const ILogicalDevice* dev)
+        : IDriverMemoryAllocation(dev, IDriverMemoryAllocation::EMAF_NONE, IDriverMemoryAllocation::EMPF_NONE), initialized(false)
+    {}
+
+    ~IOpenGLMemoryAllocation() = default;
+    
+    //! Whether the allocation was made for a specific resource and is supposed to only be bound to that resource.
+    bool isDedicated() const override { return true; }
+    
+    //! Returns the size of the memory allocation
+    size_t getAllocationSize() const override { return 0ull; }
```

Comment:
this might fuck with some of our cross-platform memory mapping code, its best if you return the buffer size for an allocation backing a buffer

(basically make `COpenGLBuffer` override it and return the same as `getSize()`)

Reply:
It already does, but since it's a pure virtual function in IDriverMemoryAllocation I had to add this in the middle. but at the end COpenGLImage/Buffer override and implement it

Reply:
But I think It was dumb, because we don't construct this class and hence I can just get rid of this